spec/integration/migration_service_spec.rb

```diff
@@ -314,6 +314,50 @@ def setup_validation_time_expectations
     end
   end
 
+  context "compressed archives" do
+    let(:old_person) {
+      FactoryBot.create(:person,
+                        profile:               FactoryBot.build(:profile),
+                        serialized_public_key: archive_private_key.public_key.export,
+                        diaspora_handle:       archive_author)
+    }
+
+    it "uncompresses gz archive" do
+      gz_compressed_file = create_gz_archive
+      service = MigrationService.new(gz_compressed_file, new_username)
+      expect(service.only_import?).to be_truthy
```

Comment:
Instead of testing `only_import?` (which has nothing to do with uncompressing) it would be better to test if `service.archive_file` returns the uncompressed file (differerent path than `gz_compressed_file` and has actual json content and isn't compressed anymore). You can use `service.send(:archive_file)` to access the private `archive_file` method from the test.

Reply:
Thanks for giving code snippets